src/mechanics/cell/Jacobian.h

```diff
@@ -33,20 +34,34 @@ class Jacobian
         }
         else
         {
-            assert (globalDimension = interpolationDimension + 1);
+            assert(globalDimension = interpolationDimension + 1);
             switch (globalDimension)
             {
             case 2:
             {
-                const int numCols = derivativeShapeFunctions.rows();
+                const int numRows = derivativeShapeFunctions.rows();
                 NodeValues nodeCopy = nodeValues;
+                // see CalculateFixedSize(...) for the Eigen::Map functionality
                 auto nodeBlockCoordinates =
-                        Eigen::Map<Eigen::Matrix<double, 2, Eigen::Dynamic>>(nodeCopy.data(), 2, numCols);
+                        Eigen::Map<Eigen::Matrix<double, 2, Eigen::Dynamic>>(nodeCopy.data(), 2, numRows);
                 mDetJacobian = (nodeBlockCoordinates * derivativeShapeFunctions).norm();
                 break;
             }
             case 3:
             {
+                const int numRows = derivativeShapeFunctions.rows();
+                NodeValues nodeCopy = nodeValues;
+                // see CalculateFixedSize(...) for the Eigen::Map functionality
+                auto nodeBlockCoordinatesMap =
+                        Eigen::Map<Eigen::Matrix<double, 3, Eigen::Dynamic>>(nodeCopy.data(), 3, numRows);
+
+                Eigen::MatrixXd nodeBlockCoordinates = nodeBlockCoordinatesMap;
```

Comment:
Can we eliminate the `nodeBlockCoordinates` object? The point of this `Eigen::Map` is to provide a _view_ to some data. So the `nodeCopy.data()` is interpreted as a `Matrix3Xd`, without a copy. Copying the `Map` into a `Matrix` somehow eliminates this advantage. Or is there a problem with the following multiplications?